code/VPC-network/security-groups.tf

```diff
@@ -0,0 +1,92 @@
+# security groups to access public (front end) and private (back end)
+# public
+resource "aws_security_group" "Wordpress-front-securitygroup" {
+ 	name = "Wordpress-front-securitygroup"
+ 	description = "front end security group - http and https inbound"
+ 	vpc_id = aws_vpc.wordpress-vpc.id
+ 	tags =	{
+ 		Name = "Wordpress-front-securitygroup"
+ 		}
+	ingress {
+		from_port = 80
+		to_port = 80
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+		ingress {
+		from_port = 443
+		to_port = 443
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+	ingress {
+		from_port = "22"
+		to_port = "22"
+		protocol = "tcp"
+		cidr_blocks = [var.MYIP]
+		}
+	egress {
+		from_port = 0
+		to_port = 0
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+	egress {
+		from_port = 443
+		to_port = 443
+		protocol = "tcp"
+		cidr_blocks = [var.vpcCIDR]
+		}
+	egress { 
+        	from_port = 3306
+        	to_port = 3306
+        	protocol = "tcp"
+        	cidr_blocks = [var.pri_subnet_cidr_2, var.pri_subnet_cidr_1]
+    }	
+}
+
+# private
+resource "aws_security_group" "Wordpress-back-securitygroup" {
+	name = "Wordpress-back-securitygroup"
+	description = "ONLY tcp CONNECTION INBOUND"
+	vpc_id = aws_vpc.wordpress-vpc.id
+	tags =	{
+		Name = "Wordpress-back-securitygroup"
+		}
+	ingress {
+		from_port = 3306
+		to_port = 3306
+		protocol = "tcp"
+		security_groups = [aws_security_group.Wordpress-front-securitygroup.id]
+		}
+	ingress {
+		from_port = "22"
+		to_port = "22"
+		protocol = "tcp"
+		cidr_blocks = [var.vpcCIDR]
+		}
+	ingress {
+		from_port = 80
+		to_port = 80
+		protocol = "tcp"
+		cidr_blocks = [var.vpcCIDR]
+		}
+	ingress {
+		from_port = 443
+		to_port = 443
+		protocol = "tcp"
+		cidr_blocks = [var.vpcCIDR]
+		}
```

Comment:
Since it is the backend where the database would be stored, I don't think that we need those ports.

Reply:
so the thought behind this was so the database server could access internet for backups storage.
https://dev.mysql.com/doc/mysql-port-reference/en/mysql-ports-reference-tables.html
i figured by making the CIDR be within the VPC, it can only push to and receive from within the VPC.

Reply:
I'd also suggest to have the security groups created on he card that will use that particular security group.
So, the SG for the database, can be created in the DB card, the ECS cluster can have the SG required for the ECS cluster, and so on so forth.
Specially if you guys will work with modules, it doesn't make sense to have a security group for a Database instance in the VPC module, if your solution  have a Database that requires a SG(like DynamoDB), otherwise you'll end up modules that can't be reused for other use cases.

Reply:
i would agree there Kiko. The reason i created them was due to the acceptance critera on the VPC card;
"No security groups or nacls have ALLOW ALL" so was trying to get them into VPC. VERY happy to remove them!